### ISSUE TITLE
Not sure if bug: query_once() creates asyncio tasks still runing in asyncio-loop after return

### ISSUE THREAD

**Issue by mirko** [cbrand/micropython-mdns#25]
I'm using `query_once()` for mDNS service-discovery at the very start of my program (and only there) and so far figured, once it returns, it's done.

However, I'm noticing - especially when hitting `CTRL+D` and therewith issuing a `KeyboardInterrupt` - that I'm sometimes interrupting running code from this very library. Way after `query_once()` was called in the first place.

The function name `query_once()` however suggests (to me) that once it returns, it's code is done. At least that's my expectation.

It seems though it creates asyncio tasks which keep running inside the asyncio loop, even after `query_once()` returned.

Is this intended or a bug? If intended, I'd suggest documenting it accordingly. Since it's calling `start()` implicitly I just wouldn't expect I have to cleanup afterwards.
Also I'd be grateful for an example ensuring that everything is properly stopped and cleaned up after I'm done with service-discovery.

**Comment by cbrand** [comment]
Hey you are right that the MDNS setup then bootstraps and runs.
As we are also always requiring to bind the MDNS socket underneath and it is expected to respond to MDNS A records "forever".

But I totally get your confusion and assumptions that you might not want that at all for service discoveries sake.

For your use case if you run `client.stop()` it will close the socket and nothing will run anymore ( the client is the Client like in the example: https://github.com/cbrand/micropython-mdns/blob/5906dd517753397d3351d9e3da6b0591d81589d8/REFERENCE.md?plain=1#L196 )

For your use case I would explicitly add a `query_once()` function ideally though which does not need to bootstrap a client object at all and also closes the socket after the execution has been done.

I also agree that this should be documented because the name `query_once()` might be confusing and definitely does not only lead you into this. Feel free to open a PR adjusting the documentation I will gladly merge it. Let me know if you cannot manage to do it and I will take it up once i have a bit of capacity.

**Comment by mirko** [comment]
Sorry for the delay - I was just about to create a PR and adding some explanation.

However while trying to come up with a good phrasing, I feel like the issue is rooted deeper.

The point is: Instantiating an object from the `Client()`-class is not creating the tasks. Instantiating from `ServiceDiscovery(client)` is.
Hence, it's really not intuitive (to me) to have to call `stop()` on the `Client`-object, as creation of the `discovery`-object leads to respective asyncio tasks. Wouldn't calling `stop()` on instances of `ServiceDiscovery()` would make more sense?

Anyway, best I came up with for the moment:

> ***Note:** Despite the call being named `query_once()`, it starts asyncio tasks which keep running -- and querying -- after the call returned. To actually stop respective background tasks, `client.stop()` needs to be called.*

**Comment by cbrand** [comment]
I did take another look and implemented a variant which stops the client automatically in query_once if it has not been manually started before.

So I removed the query_once() reference once more and made sure that the query_once actually shuts everything down if the client has not been used before. That should work also if you have a running client as it then does do nothing but implements the desired behavior of not needing to call client.stop if you haven't started it via some other means.

The changes can be found here: https://github.com/cbrand/micropython-mdns/commit/233170773933aa2de63b10cedbe3daf9392dff15

It will be available in version 1.5.2

**Comment by mirko** [comment]
Running the following code for a couple of minutes did not cause any issues and works like a charm so far!

```python
import asyncio
from mdns_client.service_discovery.txt_discovery import TXTServiceDiscovery
from mdns_client import Client as MDNSClient
import gc

sta = network.WLAN(network.STA_IF)
#sta.connect(essid, passwd)

ONLY_ONCE = True

mdns_client = None
sd = None

round = 0

def async_crash_hdlr(loop, ctx):
    print(f"ASYNCIO CRASH HANDLER CALLED IN ROUND {round}:", ctx['exception'])
    raise ctx['exception'] # trigger exit mainloop

asyncio.get_event_loop().set_exception_handler(async_crash_hdlr)

async def dns_sd():
    global mdns_client, sd
    if ONLY_ONCE or not mdns_client:
        print("Starting mDNS instance")
        mdns_client = MDNSClient(sta.ifconfig()[0])
        sd = TXTServiceDiscovery(mdns_client)
    print("Querying..")
    res = await sd.query_once("_http", "_tcp", timeout=1)
    #if ONLY_ONCE:
    #    print("Stopping mDNS instance")
    #    mdns_client.stop()
    return res

async def mainloop():
    global round
    while True:
        round += 1
        res = await dns_sd()
        print(f"Result for round {round}:", res)
        await asyncio.sleep(0)

try:
    asyncio.run(mainloop())
finally:
    print("mainloop exited")
    try:
        mdns_client.stop()
    except:
        pass
```

Thanks!
